modules/ROOT/pages/dataweave-streaming.adoc

```diff
@@ -0,0 +1,398 @@
+  },
+  {
+    "name": "Matias",
+    "age": 8
+  }
+]
+----
+
+=== Validation Failures
+
+If any of the criteria that the valiator checks are false, the validation fails. 
+
+Before proceeding, note that validation can fail in some cases when streaming works. If you write a script in a way that sequentially accesses the input variable in a given data source, streaming works, but that script might not work in all cases. For example, JSON does not place a restriction on the order of the keys in an object. If the keys in some JSON documents arrive in a different order than the script expects, streaming will fail in those cases. The annotation processor follows the rules of the format and cannot assume that the keys always arrive in the same order.
+
+==== Error: Variable is Referenced More Than Once
+
+Validation fails if a script attempts to reference the same variable more than once.
+
+The following script is designed to act on the <<example_json, JSON input>> from the <<stream_json, JSON streaming>> section. Validation fails because the script attempts to reference the `payload` variable more than once.
+
+[source,DataWeave,linenums]
+----
+%dw 2.0
+
+@StreamCapable()
+input payload application/json
+output application/json
+---
+ {
+     family: payload.family filter (member) -> member.age > 3,
+     name: payload.name
+ }
+----
+
+The script fails with the following error:
+
+[source,error,linenums]
+----
+4| input payload application/json streaming=true
+         ^^^^^^^
+Parameter `payload` is not stream capable.
+Reasons:
+ - Variable payload is referenced more than once. Locations:
+---------------------------
+
+8|      family: payload.family filter (member) -> member.age > 3,
+                ^^^^^^^
+---------------------------
+
+9|      name: payload.name
+              ^^^^^^^ at
+4| input payload application/json streaming=true
+----
+
+
+==== Error: Wrong Scope Reference
+
+Validation fails if a script attempts to reference a variable from a scope that is different from the scope in which the variable is defined.
+
+The following script fails because the `payload` variable is referenced from within the lambda expression `[1,2,3] map ((item, index) -> payload)`. Even if the expression is `[1] map ((item, index) -> payload`, streaming fails because `payload` is in the wrong scope.
+
+[source,DataWeave,linenums]
+----
+%dw 2.0
+
+@StreamCapable()
+input payload application/json
+output application/json
+---
+[1,2,3] map ((item, index) -> payload)
+---
+
+The example fails with the following error:
+
+[source,error,linenums]
+----
+4| input payload application/json
+         ^^^^^^^
+Parameter `payload` is not stream capable.
+Reasons:
+ - Variable payload is referenced in a different scope from where it was defined. Locations:
+---------------------------
+
+9| [1,2,3] map ((item, index) -> payload)
+                ^^^^^^^^^^^^^^^^^^^^^^^^ at
+4| input payload application/json
+----
+
+[[stream_output]]
+== Streaming Output
```

Comment:
Please prepare the audience to see this topic last in the set.
I thought it was a peer/option to the `streaming` parameter and expected to see it earlier.